core/src/test/java/com/datastax/oss/driver/internal/core/context/DefaultDriverContextTest.java

```diff
@@ -0,0 +1,93 @@
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.datastax.oss.driver.api.core.config.DefaultDriverOption;
+import com.datastax.oss.driver.api.core.config.DriverExecutionProfile;
+import com.datastax.oss.driver.internal.core.protocol.Lz4Compressor;
+import com.datastax.oss.driver.internal.core.protocol.SnappyCompressor;
+import com.datastax.oss.protocol.internal.Compressor;
+import com.datastax.oss.protocol.internal.NoopCompressor;
+import io.netty.buffer.ByteBuf;
+import java.util.Optional;
+import org.junit.Test;
+
+public class DefaultDriverContextTest {
+
+  private DefaultDriverContext buildMockedContext(Optional<String> compressionOption) {
+
+    DriverExecutionProfile defaultProfile = mock(DriverExecutionProfile.class);
+    when(defaultProfile.isDefined(DefaultDriverOption.PROTOCOL_COMPRESSION))
+        .thenReturn(compressionOption.isPresent());
+    compressionOption.ifPresent(
+        compression -> {
+          when(defaultProfile.getString(DefaultDriverOption.PROTOCOL_COMPRESSION))
+              .thenReturn(compression);
+        });
+    return MockedDriverContextFactory.defaultDriverContext(Optional.of(defaultProfile));
+  }
+
+  private void doCreateCompressorTest(Optional<String> configVal, Class<?> expectedClz) {
+
+    DefaultDriverContext ctx = buildMockedContext(configVal);
+    Compressor<ByteBuf> compressor = ctx.getCompressor();
+    assertThat(compressor).isNotNull();
+    assertThat(compressor).isInstanceOf(expectedClz);
+  }
+
+  @Test
+  public void should_create_lz4_compressor() {
+
+    doCreateCompressorTest(Optional.of("lz4"), Lz4Compressor.class);
+
+    /* Config should be case-insensitive */
+    doCreateCompressorTest(Optional.of("lZ4"), Lz4Compressor.class);
+    doCreateCompressorTest(Optional.of("Lz4"), Lz4Compressor.class);
+    doCreateCompressorTest(Optional.of("LZ4"), Lz4Compressor.class);
+  }
+
+  @Test
+  public void should_create_snappy_compressor() {
+
+    doCreateCompressorTest(Optional.of("snappy"), SnappyCompressor.class);
+
+    /* Config should be case-insensitive */
+    doCreateCompressorTest(Optional.of("SNAPPY"), SnappyCompressor.class);
+    doCreateCompressorTest(Optional.of("sNaPpY"), SnappyCompressor.class);
+    doCreateCompressorTest(Optional.of("SNapPy"), SnappyCompressor.class);
+  }
+
+  @Test
+  public void should_create_noop_compressor_if_undefined() {
+
+    doCreateCompressorTest(Optional.empty(), NoopCompressor.class);
+  }
+
+  @Test
+  public void should_create_noop_compressor_if_defined_as_none() {
+
+    doCreateCompressorTest(Optional.of("none"), NoopCompressor.class);
+
+    /* Config should be case-insensitive */
+    doCreateCompressorTest(Optional.of("NONE"), NoopCompressor.class);
+    doCreateCompressorTest(Optional.of("NoNe"), NoopCompressor.class);
+    doCreateCompressorTest(Optional.of("nONe"), NoopCompressor.class);
```

Comment:
This (and the similar cases above) are a great argument for generative testing... but per the conversation with @tomekl007 elsewhere re: parameterized tests I didn't want to introduce a new dependency just for this case.  I'm happy to look at this more deeply if something like that is desired, though.

Reply:
We should use `com.tngtech.java.junit.dataprovider`, see my comment:
https://github.com/datastax/java-driver/pull/1336#discussion_r329432605

Reply:
It's not exactly equivalent to generative testing... but switching to the dataprovider seems to strike a pretty reasonable middle ground.

Reply:
Yeah we've not explored generative testing on the project, I'm not sure if we have many good cases for it...
In this particular case I think it would be a bit overkill.